crates/net/network/src/peers/manager.rs

```diff
@@ -378,20 +379,33 @@ impl PeersManager {
                 })
             }
         } else {
-            let reputation_change = if let Some(kind) = err.should_backoff() {
-                // The peer has signaled that it is currently unable to process any more
-                // connections, so we will hold off on attempting any new connections for a while
-                self.backoff_peer(*peer_id, kind);
-                BACKOFF_REPUTATION_CHANGE.into()
-            } else {
-                self.reputation_weights.change(reputation_change)
-            };
+            let mut backoff_until: Option<std::time::Instant> = None;
 
             if let Some(mut peer) = self.peers.get_mut(peer_id) {
+                let reputation_change = if let Some(kind) = err.should_backoff() {
+                    let backoff_time =
+                        self.backoff_durations.backoff_until(kind, peer.backoff_counter);
+
+                    backoff_until = Some(backoff_time);
+
+                    // Increment peer.backoff_counter
+                    peer.backoff_counter += 1;
```

Comment:
if we increment afterwards and start at `0`, this will be 0, right?

I'd also love to see a test for that, see existing tests for ref

Reply:
>if we increment afterwards and start at 0, this will be 0, right?

It won't be zero because we add one in the `backoff_until` function. I can move the increment before the calculation that way we can remove the + 1 in `backoff_until`. Please let me know which is better.

> I'd also love to see a test for that, see existing tests for ref

On it